### DISCUSSION TITLE
Difference: CODE and code + BTs and codelists

### DISCUSSION THREAD

**Discussion by troa1** [OP-TED/eForms-SDK#82] [Q&A] [answered]
I have a few questions regarding codelists - has someone found answers to these in the SDK or elsewhere??

1. In the fields.json (fields.legalType and fields.type) word code is written in two different ways **CODE** and **code** (in small letters). What is the difference between these two? I can see that sometimes these two are both CODE and code but sometimes the are different e.g. INDICATOR and code (see BT-106-Procedure)? In the case of BT-106 there is a codelists indicated in fields.xpathAbsolute (@listName='accelerated-procedure') eventhough in fields.legalType this should be INDICATOR.

2. In fields.json (fields.xpathAbsolute) some of the BT's have the name of **codelists that should be associated with the BT.** But not all have this. E.g.  BT-771-Lot has @listName='missing-info-submission' but  BT-783-Review does not have a codelist indicated in any of the fields (this is marked as CODE and code in fields.legalType and fields.type). **Has someone found a full list of codes + in which BT is to be used?**  [e.g. BT-01-notice -> legal.basis]?

**Comment by jpmckinney** [comment, marked as answer]
`legalType` is [defined](https://docs.ted.europa.eu/eforms/1.0.0/fields/index.html) as "The data type of the Business Term associated with the field as defined by the eForms Regulation." and as "Data type of the business term, as indicated in the eForms Regulation." and `type` as "The data type of the field. [See Field data types](https://docs.ted.europa.eu/eforms/1.0.0/fields/index.html#data-types)." and as "Technical data type of the field.".

In short, `legalType` is the type at the level of the model/ontology, and `type` is the type at the level of the UBL/XML implementation.

> but BT-783-Review does not have a codelist indicated in any of the fields

I see it is `review-status`. You need to look at the `codeList` property of the field, not the `xpathAbsolute` property:

```json
    "codeList" : {
      "value" : {
        "id" : "review-status",
        "type" : "flat"
      },
      "severity" : "ERROR"
    }
```
